testing/sliding-sync-integration-test/src/notification_client.rs

```diff
@@ -0,0 +1,209 @@
+
+use anyhow::{ensure, Result};
+use assert_matches::assert_matches;
+use matrix_sdk::{
+    config::SyncSettings,
+    ruma::{
+        api::client::room::create_room::v3::Request as CreateRoomRequest,
+        assign,
+        events::{
+            room::{member::MembershipState, message::RoomMessageEventContent},
+            AnyStrippedStateEvent, SyncMessageLikeEvent, TimelineEventType,
+        },
+        OwnedEventId,
+    },
+    RoomState,
+};
+use matrix_sdk_integration_testing::helpers::get_client_for_user;
+use matrix_sdk_ui::notification_client::{
+    Error, NotificationClient, NotificationEvent, NotificationItem, NotificationStatus,
+};
+use tracing::warn;
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+async fn test_notification() -> Result<()> {
+    // Create new users for each test run, to avoid conflicts with invites existing
+    // from previous runs.
+    let time = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_millis();
+    let alice = get_client_for_user(format!("alice{time}"), true).await?;
+    let bob = get_client_for_user(format!("bob{time}"), true).await?;
+
+    // Alice changes display name.
+    const ALICE_NAME: &str = "Alice, Queen of Cryptography";
+    alice.account().set_display_name(Some(ALICE_NAME)).await?;
+
+    // Initial setup: Alice creates a room, invites Bob.
+    let invite = vec![bob.user_id().expect("bob has a userid!").to_owned()];
+    let request = assign!(CreateRoomRequest::new(), {
+        invite,
+        is_direct: true,
+    });
+
+    let alice_room = alice.create_room(request).await?;
+
+    const ROOM_NAME: &str = "Kingdom of Integration Testing";
+    alice_room.set_name(Some(ROOM_NAME.to_owned())).await?;
+
+    let room_id = alice_room.room_id().to_owned();
+
+    // Bob receives a notification about it.
+
+    let bob_invite_response = bob.sync_once(Default::default()).await?;
+    let sync_token = bob_invite_response.next_batch;
+
+    let mut invited_rooms = bob_invite_response.rooms.invite.into_iter();
+
+    let (_, invited_room) = invited_rooms.next().expect("must be invited to one room");
+    assert!(invited_rooms.next().is_none(), "no more invited rooms: {invited_rooms:#?}");
+
+    if let Some(event_id) = invited_room.invite_state.events.iter().find_map(|event| {
+        let Ok(AnyStrippedStateEvent::RoomMember(room_member_ev)) = event.deserialize() else {
+            return None;
+        };
+
+        if room_member_ev.content.membership != MembershipState::Invite {
+            return None;
+        }
+
+        let Ok(Some(event_id)) = event.get_field::<OwnedEventId>("event_id") else {
+            return None;
+        };
+
+        Some(event_id)
+    }) {
+        warn!("We found the invite event!");
+
+        // Try with sliding sync first.
+        let notification_client = NotificationClient::builder(bob.clone()).await.unwrap().build();
+        let NotificationStatus::Event(notification) =
```

Comment:
We generally prefer using `assert_matches!()` for these types of assertions. Can we do so here as well?